Enable native pivot xlsx and Incrementally add rows to Native Pivot xlsx exports

adam-james-v opened the pull request:
This PR adds a boolean env var `MB_TEMP_NATIVE_PIVOT_EXPORTS` which enables the 'native pivot' feature in Downloads of xlsx files.

Targeting the 50 branch only for now, as pivot export improvements are being added holistically in 51.
Relevant Slack thread discussing this:

https://metaboat.slack.com/archives/C01LQQ2UW03/p1724874779867859

adam-james-v committed:
Incrementally add rows to Native Pivot xlsx exports

Adds a boolean env var `MB_NATIVE_PIVOT_EXPORTS` which enables the 'native pivot' feature in Downloads of xlsx files.

adam-james-v committed:
Simplify the init

adam-james-v committed:
remove unnecessary step

metamben reviewed (approved):
LGTM When `row-count-estimate` is calculated, we should make sure that `max` gets at least one number. (It might already be the case, difficult for me to tell.)

adam-james-v committed:
properly use streaming workbook for pivot export too

This change initializes a non-streaming workbook so that a Pivot table can be added. Then, it turns that workbook into
a streaming workbook, so that we never have to keep the entire output in memory; just like non-pivot xlsx exports, we
can ensure more efficient output.

adam-james-v commented on src/metabase/query_processor/streaming/xlsx.clj:
```diff
@@ -565,11 +564,11 @@
+  [{:keys [pivot-grouping-key] :as pivot-spec}
    {:keys [ordered-cols col-settings viz-settings]}]
-  (let [idx-shift                   (fn [indices]
+  (let [row-count-estimate          (apply max (remove nil? (map #(get-in % [:fingerprint :global :distinct-count]) ordered-cols)))
```
This is a good catch. 
I've changed this function so no row count estimate is needed at all, so this won't be an issue.

adam-james-v committed:
Env var name more indicative of its temporary nature

I mean, all things eventually succumb to the inexorable march of time that sands away all lustre, it's just that this
environment variable will disappear much sooner, and intentionally.

adam-james-v committed:
change env var name, add default row count in case no counts exist

adam-james-v committed:
try to improve the area-ref code

adam-james-v committed:
just kidding, figured out how to do A:D style refs

Still need to avoid the zip bomb problem, but at least now we don't need to know or guess the row count beforehand.

And, the fileszie ends up being smaller, though the time to download goes up a little bit.

adam-james-v committed:
Merge branch 'release-x.50.x' into enable-native-pivot-xlsx

adam-james-v committed:
fix xlsx test

adam-james-v committed:
should keep col name format behaviour the same
